Added support for pipenv and non-standard python configurations

tiagoad opened the pull request:
This pull-request makes this project compatible with pipenv.

It can be run with

```bash
pipenv run ./vyos-build-ami "<image_uri>"
```

I'm not sure if the changes will break regular usage, but they shouldn't as I believe it will default to use the default PATH `python`, instead of /bin/python (which was breaking for me on Mac OS)

Commit:
added support for pipenv and non-standard python configurations

tiagoad commented:
Actually, this forces Python version 3.7. Removing the last section in the Pipfile should fix those problems.

dmbaturin commented:
Thanks! Looks good at a glance, I'll try it.

Have you made a working AMI by the way? I've fixed the readme to mention the "make AWS" target required for the ISO build to include the EC2 initialization stuff.

That cloud decoupling is a relatively recent change, so the readme was out of sync with reality and lying by omission!

tiagoad commented:
Also, there's a missing dep. I'll send another pull-request in a few minutes, as soon as I'm sure it's working.

tiagoad commented:
>Have you made a working AMI by the way? I've fixed the readme to mention the "make AWS" target required for the ISO build to include the EC2 initialization stuff.

Yes, I've modified the playbooks to use debian scratch (and necessarily overlay instead of aufs) but I still haven't confirmed it's working. I'm hoping to have it working before the end of the day.
